### ISSUE TITLE
[cmake] run tests with python3 (Trac #1777)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1769]
[cmake] run tests with python3. when adding a test that is a python script, prepend the cmake python executable to the command. this should allow testing with python3. owner: nega

Migrated from https://code.icecube.wisc.edu/ticket/1777
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:13:10", 
    "description": "[cmake] run tests with python3. when adding a test that is a python script, prepend the cmake python executable to the command. this should allow testing with python3. owner: nega", 
    "reporter": "david.schultz", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067190995086", 
    "component": "cmake", 
    "summary": "[cmake] run tests with python3", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2016-07-08T15:34:39", 
    "milestone": "", 
    "owner": "nega", 
    "type": "rumor / allegation"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160708T17:44:24: 
* nega changed status from "new" to "closed"
* nega commented:

> In changeset 2623:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2623"
> run tests as "python test.py" not "./test.py"
> 
> if a test added via `i3_test_scripts()` ends with the regex "\.py$", assume
> it is a python script and run it with the value of `${PYTHON_EXECUTABLE}` as
> detected in `cmake/tools/python.cmake`. this should ease using python3.
> 
> closes #1769
> ```


* nega changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20190112T00:22:18: nwhitehorn commented:

> In changeset 170295:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="170295"
> run tests as "python test.py" not "./test.py"
> 
> if a test added via `i3_test_scripts()` ends with the regex "\.py$", assume
> it is a python script and run it with the value of `${PYTHON_EXECUTABLE}` as
> detected in `cmake/tools/python.cmake`. this should ease using python3.
> 
> closes #1769
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:13:10: 
* olivas commented:

> Milestone July 2016 deleted


* olivas changed milestone from "July 2016" to ""
